app.js

```diff
@@ -1,8 +1,19 @@
 const bodyParser = require('body-parser');
 const express = require('express');
-
+const mongoose = require('mongoose');  // using ODM wrapper for MongoDB
+const database = require('./config/database');
+const morgan = require('morgan');  // log requests to the console
 const app = express();
 
+// set up MongoDB connection using the global promise library and then get connection
+mongoose.connect(database.url, {promiseLibrary: global.Promise}, error => {
+  if (error) {
+    console.log(`MongoDB connection error: ${error}`);
+    process.exit(1);  // should consider alternative to exiting the app due to db conn issue
+  }
+});
+const db = mongoose.connection;
+
```

Comment:
the code here is fine just gonna throw this here as example of a different method cause it's interesting to see other soultions
 `// connect to mongoDB

mongoose.connect(database.url);
const db = mongoose.connection;

// Check connection
db.once('open', function(){
    console.log('Connected to MongoDB');
});

// Check for db errors
db.on('error', function(err){
    console.log(err);
});`